feat: Homepage

bzgaspar opened the pull request:
Fetch all countries from API. Display country using country card. add search & filter region. Since the api does not support pagination I create a inifite scroll feature to display initial countries then load more when scrolled down

bzgaspar committed:
feat: Homepage

Fetch all countries from API. Display country using country card. add search & filter region (The search and filter functionality are in different ticket)

eremocal commented on src/components/Header/SubComponents/ToggleDarkMode.tsx:
```diff
@@ -0,0 +1,21 @@
+      onClick={toggleTheme}
+      className={Style['dark-mode-button']}
+      type='button'
+      aria-label='Dark Mode Button'
```
I don't think we need aria-label here

eremocal commented on src/components/common/form-control/DropdownSelect.tsx:
```diff
@@ -0,0 +1,56 @@
+        className={Style.dropdown__toggle}
+      >
+        {props.toggleLabel}
+        <svg
```
is it possible to move the svg to it's own file?

eremocal reviewed (commented):
![image](https://github.com/itsmeNEK/rest-countries-api/assets/61583701/21b3b0d7-61bc-4e44-875e-c490dbd4618b)
Please add spacing on left and right of the layout

**edit: fixed**

bzgaspar committed:
feat: Homepage

add pagination on counrties(12 each page).  Create arrowdown svg in seperate file. Remove aria-label in toggle dark mode.

eremocal reviewed (commented):
![image](https://github.com/itsmeNEK/rest-countries-api/assets/61583701/0493a0df-f68f-4fa7-b553-c546d643bd9c)
please add spacing on the layout

**edit: fixed**

eremocal commented on src/components/Countries/PaginateCountries.tsx:
```diff
@@ -0,0 +1,52 @@
+  return (
+    <>
+      {paginatedData.map((country: CountryTypes, index: number) => (
+        <CountryCard key={index} country={country} />
```
![image](https://github.com/itsmeNEK/rest-countries-api/assets/61583701/266492ac-863b-4daf-ad7f-6fd245fddd86)
please add a margin bottom at the end of the countries

eremocal commented on src/app/page.tsx:
```diff
@@ -1,21 +1,14 @@
+}: {
+  [key: string]: string | string[] | undefined
+}) {
+  console.log(searchParams)
```
please remove the log here

eremocal commented on src/components/Countries/SubComponent/CountryCard/CountryCard.tsx:
```diff
@@ -10,17 +10,17 @@ type CountryCardProps = {
         />
         <div className={Style['card__body']}>
-          <h1 className={Style['card__body__title']}>{country.name}</h1>
+          <h1 className={Style['card__body__title']}>{country.name.common}</h1>
```
there should be only one h1's on the page

bzgaspar committed:
feat: Homepage

Change h1 to p in country card. Remove log in page. Adjust space in layout

bzgaspar committed:
feat: Change p to h2. correct margin-top to margin(add space at the bottom of the country cards)
